src/main/java/com/epam/esm/service/CertificateService.java

```diff
@@ -0,0 +1,135 @@
+package com.epam.esm.service;
+
+import com.epam.esm.dao.CertificateDAO;
+import com.epam.esm.dto.CertificateRequestModel;
+import com.epam.esm.dto.CertificateResponseModel;
+import com.epam.esm.dto.TagRequestModel;
+import com.epam.esm.dto.TagResponseModel;
+import com.epam.esm.entity.Certificate;
+import com.epam.esm.exception.ResourceNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class CertificateService {
+    private final CertificateDAO certificateDAO;
+    private final ModelMapper modelMapper;
+    private final TagService tagService;
+    private final CertificateHasTagService certificateHasTagService;
+
+    @Transactional
+    public void create(CertificateRequestModel certificateRequestModel) {
+        Certificate certificate = modelMapper.map(certificateRequestModel, Certificate.class);
+        String date = LocalDateTime.now().toString();
+        certificate.setCreateDate(date);
+        certificate.setLastUpdateDate(date);
+        List<TagRequestModel> newTagRequestModels = certificateRequestModel.getTagRequestModels();
+        List<TagResponseModel> actualTagRequestModels = tagService.getTags();
+
+        List<TagRequestModel> addTags = newTagRequestModels
+                        .stream()
+                        .filter(t1 -> actualTagRequestModels.stream().noneMatch(t2 -> t1.getName().equals(t2.getName()))).collect(Collectors.toList());
+
+        addTags.forEach(tagService::create);
```

Comment:
Please consider formating this line horizontally
also you can add .forEach call to the same chain